core/src/main/java/org/infinispan/xsite/irac/DefaultIracManager.java

```diff
@@ -531,4 +557,103 @@ synchronized boolean isCompleted() {
          return stateStatus == StateStatus.COMPLETED;
       }
    }
+
+   @ManagedAttribute(description = "Number of keys that need to be sent to remote site(s)", displayName = "XsiteQueueSize", measurementType = MeasurementType.DYNAMIC)
+   public int getQueueSize() {
+      return updatedKeys.size();
```

Comment:
checking `getStatisticsEnabled()` is missing. should return `-1` if stats are disabled.

Reply:
Push the changes